move incl/phi/prior_std calculation in preprocess

mkanai opened the pull request:
I moved incl/phi/prior_std calculation to the preprocess task so that they won't be computed for every region.

@Fedja re: `is_binary` flag that we discussed, I don't know it's necessary because we now hard-coded 0/1 in L71 anyways. When binary phenotypes are coded other than 0/1/NA, is there any simple way to infer binary codings?

Fedja commented:
> I moved incl/phi/prior_std calculation to the preprocess task so that they won't be computed for every region.
> 
> @Fedja re: `is_binary` flag that we discussed, I don't know it's necessary because we now hard-coded 0/1 in L71 anyways. When binary phenotypes are coded other than 0/1/NA, is there any simple way to infer binary codings?

it would be to protect the users from themselves but not necessary off course. is_binary would check that the values are 0,1,NA and nag if not.
say user accidentally codes cases/controls with 1/2 or happens to have one erroneus value there which triggers quantitative. At least could add bolded notification in readme that if there are values other than 1,0,NA quant pheno is assumed.

Fedja commented on wdl/finemap.wdl:
```diff
@@ -30,7 +31,58 @@ task preprocess {
+                phi = vals["1"] / (vals["1"]+vals["0"])
+                var_y = phi * (1-phi)
+                std = 0.05 * sqrt(phi*(1-phi))
+                printf var_y > "var_y.txt"
```
printf here and not others does not affect anything except missing newline but looks suspicious to autist like me :)

mkanai replied:
great catch! addressed

mkanai commented:
> > I moved incl/phi/prior_std calculation to the preprocess task so that they won't be computed for every region.
> > @Fedja re: `is_binary` flag that we discussed, I don't know it's necessary because we now hard-coded 0/1 in L71 anyways. When binary phenotypes are coded other than 0/1/NA, is there any simple way to infer binary codings?
> 
> it would be to protect the users from themselves but not necessary off course. is_binary would check that the values are 0,1,NA and nag if not. say user accidentally codes cases/controls with 1/2 or happens to have one erroneus value there which triggers quantitative. At least could add bolded notification in readme that if there are values other than 1,0,NA quant pheno is assumed.

I agree the flag would prevent mistakes but it also prevent fine-mapping binary and quantitative traits altogether. I added a bold note in README for now.

Fedja commented on README.md:
```diff
@@ -19,7 +19,8 @@ Please run [`finemap.wdl`](wdl/finemap.wdl) on a cromwell server with an appropr
 - `finemap.sumstats_pattern`: a path to GWAS summary statistics where `{PHENO}` is a magic keyword to be replaced by an actual phenotype name in `finemap.phenolistfile`
-- `finemap.phenotypes`: a list of phenotypes to fine-map
+- `finemap.phenolistfile`: a list of phenotypes to fine-map
+- `finemap.phenotypes`: a path to phenotype file. **[IMPORTANT]** The first column should contains sample IDs that match to bgen sample IDs. If a phenotype is binary, we expect it's coded as **0/1/NA**.
```
continue like: ....coded 0/1/NA, otherwise quantitative phenotype is automatically assumed

mkanai replied:
addressed!

Lipastomies reviewed (approved):
Looking good! I agree with the assumption of a certain data form for binary data, as long as it's documented clearly it should not be a problem.
